addons/account/models/account_move.py

```diff
@@ -1134,15 +1135,32 @@ def _query_get(self, domain=None):
         date_field = 'date'
         if context.get('aged_balance'):
             date_field = 'date_maturity'
-        if context.get('date_to'):
-            domain += [(date_field, '<=', context['date_to'])]
-        if context.get('date_from'):
-            if not context.get('strict_range'):
-                domain += ['|', (date_field, '>=', context['date_from']), ('account_id.user_type_id.include_initial_balance', '=', True)]
-            elif context.get('initial_bal'):
-                domain += [(date_field, '<', context['date_from'])]
-            else:
-                domain += [(date_field, '>=', context['date_from'])]
+
+        if context.get('companies_fiscalyears'):
+            sub_domains = []
+            for v in context['companies_fiscalyears']['companies']:
+                date_from = v.get('date_from')
+                date_to = v['date_to']
+                sub_domain = [('company_id', '=', v['company_id']), (date_field, '<=', date_to)]
+                if date_from:
+                    if not context.get('strict_range'):
+                        sub_domain += ['|', (date_field, '>=', date_from), ('account_id.user_type_id.include_initial_balance', '=', True)]
+                    elif context.get('initial_bal'):
+                        sub_domain += [(date_field, '<', date_from)]
+                    else:
+                        sub_domain += [(date_field, '>=', date_from)]
+                sub_domains.append(sub_domain)
+            domain += expression.OR(sub_domains)
+        else:
+            if context.get('date_to'):
+                domain += [(date_field, '<=', context['date_to'])]
+            if context.get('date_from'):
+                if not context.get('strict_range'):
+                    domain += ['|', (date_field, '>=', context['date_from']), ('account_id.user_type_id.include_initial_balance', '=', True)]
+                elif context.get('initial_bal'):
+                    domain += [(date_field, '<', context['date_from'])]
+                else:
+                    domain += [(date_field, '>=', context['date_from'])]
```

Comment:
I think the code and the diff would be clearer by defining a small helper function inside the query_get() method:
```python
def _compute_date_domain(date_field, date_from, date_to, company_id=False):
    domain = []
    if company_id:
        domain += [('company_id', '=', company_id)
    if date_to:
        domain += [(date_field, '<=', date_to)]
    if date_from:
        if not context.get('strict_range'):
            domain += ['|', (date_field, '>=',date_from), ('account_id.user_type_id.include_initial_balance', '=', True)]
        elif context.get('initial_bal'):
            domain += [(date_field, '<', date_from)]
        else:
            domain += [(date_field, '>=', date_from)]
    return domain
```